text/0328-polyglot-assert.md

```diff
@@ -0,0 +1,335 @@
+---
+rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/330
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/328
+---
+
+# Polyglot Assert
+
+The `@aws-cdk/assert` module is only available to javascript users.
+
+The new polyglot assert module - `@aws-cdk/assertv2` - provides the same set of functionalities
+to users in all CDK supported languages.
+
+## Working Backwards
+
+### CHANGELOG
+
+**feat:** announcing assertv2: the assert library is now available in all CDK supported languages.
+
+### README
+
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![cdk-constructs: Stable](https://img.shields.io/badge/cdk--constructs-stable-success.svg?style=for-the-badge)
+
+---
+
+<!--END STABILITY BANNER-->
+
+> NOTE: This module contains *beta APIs*.
+>
+> All beta symbols are suffixed with the `Beta<n>`. When we have backwards
+> incompatible change, we will create a new symbol with a `Beta<n+1>` suffix
+> and deprecate the `Beta<n>` symbol.
+---
+This module allows asserting the contents of CloudFormation templates.
+
+To run assertions based on a CDK `Stack`, start off with -
+
+```ts
+const stack = new cdk.Stack(...)
+...
+const inspect = StackAssertionsBeta1.fromStackBeta1(stack);
```

Comment:
> Let's write this README file as if the API was final (without the pre-release postfixes), so we can sense the ergonomics without the "pollution".

I've intentionally taken the opposite approach to get a sense of the ergonomics with the pollution. This is the APIs our users would encounter initially. 
I assumed it's not too hard to imagine the API without the pollution compared to how it would be with the "pollution".

> As for the name of the entrypoint, StackAssertions is okay, but I am wondering if this is more of TemplateAssertions. It "speaks" in CFN template terminology, not in CDK stack terms. WDTY?

I like it.

> I am also wondering if we should use Preview instead of Beta to avoid confusion

I believe we're already starting to use this - https://github.com/aws/aws-cdk/pull/13780
Happy to change, of course. cc @NetaNir @skinny85

Reply:
I am okay with `beta1` (just on the top class of course)

Reply:
Yes, `Preview` was "voted off the island" in https://github.com/aws/aws-cdk/pull/13780#pullrequestreview-629595262.

Reply:
Yup `betaX` it is. Adam and I will figure out a way to announce it and document it asap